http3/server.go

```diff
@@ -248,6 +274,17 @@ func (s *Server) handleRequest(sess quic.Session, str quic.Stream, decoder *qpac
 		s.logger.Infof("%s %s%s", req.Method, req.Host, req.RequestURI)
 	}
 
+	// Now set the write timeout
+	t0 = time.Now()
+	if d := s.WriteTimeout; d != 0 {
```

Comment:
```
// WriteTimeout is the maximum duration before timing out
// writes of the response. It is reset whenever a new
// request's header is read. Like ReadTimeout, it does not
// let Handlers make decisions on a per-request basis.
```

I'm not sure I fully understand this, but don't we need to reset the deadline when a new header is read?

Reply:
I reset the timeout at the end of handleRequest. I have the feeling that it may not be necessary cause the stream will be closed after handleRequest. But it seems safer in this way.

Reply:
I was referring to
```
It is reset whenever a new request's header is read.
```

Does that mean we need to reset the deadline when the server receives a new request?

Reply:
We are not reusing stream, are we? The server in handleConn, accepts a Stream, handles it and then closes it. I believe this phrase make more sense for http2 which handles multiple requests in single connection. Please correct me if I'm wrong.

Reply:
We're not reusing streams, and HTTP/2 is neither. As HTTP/2, HTTP/3 also handles multiple requests on a single connection.

Reply:
Correct. The deadline that we set here is stream level though.

Let me research more Marten.

Reply:
Thank you @ohoushayr! I'm not sure how it's supposed to work, just reading the Godoc here. I guess the best thing would be to see how this is implemented in Go's HTTP/2 implementation.

Reply:
In HTTP/2 at the beginning of ServeConn, session level write timeout will be disarmed. Then the stream level write timeout will be set after reading header frame, at newStream. It will never reset. In stream closing however, the write timeout timer will be reset.

The Godoc is only talks about HTTP/1.x which wouldn't be completely applicable to our implementation.